manifests/vrrp/unicast_peer.pp

```diff
@@ -0,0 +1,24 @@
+# == Define: keepalived::vrrp::unicast_peer
+#
+# === Parameters:
+#
+# $name::     IP address of unicast peer
+# $instance:: Name of vrrp instance this peer belongs to
+#
+define keepalived::vrrp::unicast_peer (
+  $instance,
+) {
+  assert_private()
+
+  $_inst = regsubst($instance, '[:\/\n]', '')
+
+  validate_ip_address($name)
```

Comment:
I prefer `assert_type($name, Stdlib::IP::Address)` instead of calling `validate_ip_address()`. The first is in Puppet core, where the latter requires stdlib.

We can even make this easier by adding a parameter to the defined resource:

```puppet
Stdlib::IP::Address $ip = $name,
```